packages/app-mobile/components/NoteEditor/SearchPanel.tsx

```diff
@@ -0,0 +1,335 @@
+// Displays a find/replace dialog
+
+const React = require('react');
+const { StyleSheet } = require('react-native');
+const { TextInput, View, Text, TouchableOpacity } = require('react-native');
+const { useMemo, useState, useEffect } = require('react');
+const MaterialCommunityIcon = require('react-native-vector-icons/MaterialCommunityIcons').default;
+
+import { SearchControl, SearchState, EditorSettings } from './types';
+import { _ } from '@joplin/lib/locale';
+import { BackHandler } from 'react-native';
+
+const buttonSize = 48;
+
+export const defaultSearchState: SearchState = {
+	useRegex: false,
+	caseSensitive: false,
+
+	searchText: '',
+	replaceText: '',
+	dialogVisible: false,
+};
+
+export interface SearchPanelProps {
+    searchControl: SearchControl;
+    searchState: SearchState;
+    editorSettings: EditorSettings;
+}
+
+const ActionButton = (
+	props: { styles: any; iconName: string; title: string; onPress: (()=> void) }
+) => {
+	return (
+		<TouchableOpacity
+			style={props.styles.button}
+			onPress={props.onPress}
+
+			accessibilityLabel={props.title}
+			accessibilityRole='button'
+		>
+			<MaterialCommunityIcon name={props.iconName} style={props.styles.buttonText}/>
+		</TouchableOpacity>
+	);
+};
+
+const ToggleButton = (
+	props: { styles: any; iconName: string; title: string; active: boolean; onToggle: (()=> void) }
+) => {
+	const active = props.active;
+
+	return (
+		<TouchableOpacity
+			style={{
+				...props.styles.toggleButton,
+				...(active ? props.styles.toggleButtonActive : {}),
+			}}
+			onPress={props.onToggle}
+
+			accessibilityState={{
+				checked: props.active,
+			}}
+			accessibilityLabel={props.title}
+			accessibilityRole='switch'
+		>
+			<MaterialCommunityIcon name={props.iconName} style={
+				active ? props.styles.activeButtonText : props.styles.buttonText
+			}/>
+		</TouchableOpacity>
+	);
+};
+
+
+const useStyles = (theme: any) => {
```

Comment:
This should have the `Theme` type (unless it's not possible to import it here for some reason).

Reply:
I wasn't aware that type existed! Good catch!